dhall-mode.el

```diff
@@ -177,7 +183,8 @@ down.  You can also disable type-checking entirely by setting
 
 (reformatter-define dhall-format
   :program (or dhall-format-command dhall-command)
-  :args (unless dhall-format-command '("format"))
+  :args (let ((cmd (if dhall-format-with-ascii '("--ascii") '())))
+          (if dhall-format-command cmd (cons "format" cmd)))
```

Comment:
this could instead be something like
```el
(concat (unless dhall-format-command '("format"))
        (when dhall-format-with-ascii '("--ascii")))
```

Reply:
That's much nicer, cheers.

Reply:
Arrived at:
```lisp
(append (unless dhall-format-command '("format"))
        (when dhall-format-with-ascii '("--ascii")))
```
I wondered about defining argument list as you suggest above. As `--ascii` is the only meaningful argument for the formatter (i.e. `--inplace` and `--help` wouldn't make sense for the reformatter), the boolean flag seemed like a friendlier interface.

Reply:
Resting on the assumption that `dhall format` remains opinionated. Agree this could be less convenient in future, so would bow to your better judgement.

Reply:
Happy to leave it as-is, but would you mind renaming the flag to `dhall-format-ascii`? I think the `with` is a bit redundant.

Reply:
In fact, thinking about it, I think I still prefer the `dhall-format-arguments` approach. As soon as a `--width=80` option or similar appears, managing all the flags becomes. There's no reason for us to try to provide structured wrappers for these options unless they are relevant to logic in the elisp, particularly as the options change over time with the versions of the wrapped tool. A docstring can show `'("--ascii")` as an example.

Reply:
Makes sense, it's more flexible this way for sure :+1: